src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor_Minimal.cs

```diff
@@ -166,7 +165,24 @@ private void MinimallyQualify(INamedTypeSymbol symbol)
                 return SpecializedCollections.EmptyDictionary<INamespaceOrTypeSymbol, IAliasSymbol>();
             }
 
-            var token = semanticModelOpt.SyntaxTree.GetRoot().FindToken(positionOpt);
+            // Walk up the ancestors from the current position. If this is a speculative
+            // model, walk up the corresponding ancestors in the parent model.
+            SemanticModel semanticModel;
+            int position;
+            if (semanticModelOpt.IsSpeculativeSemanticModel)
+            {
+                var methodRoot = ((MethodBodySemanticModel)semanticModelOpt).Root;
+                Debug.Assert(methodRoot.FullSpan.Contains(positionOpt));
+                semanticModel = semanticModelOpt.ParentModel;
+                position = methodRoot.FullSpan.Start;
```

Comment:
It is not obvious why there should be a dependency in position between two unrelated syntax trees.